### PR TITLE
I003-A - Website Theme

### PR DESCRIPTION
**Pull request by AZZohaib**
image
image
image
image
image
image
image
image

### CODE REVIEW

**Review by Adnanhoq — DISMISSED** [review]

**Review by Adnanhoq — DISMISSED** [review]

File: views/components/footer.njk [hunk 1]
```diff
@@ -1,12 +1,28 @@
-<footer class="d-flex fixed-bottom flex-wrap justify-content-between align-items-center py-3 border-top" style="background-color:white">
-    <p class="col-md-4 mb-0">© 2024 Kainos</p>
-    
-    <div class="nav-logo-container">
-        <img src="../assets/index/kainos-logo.png" alt="transparent kainos logo for the navbar" class="img-fluid">
-    </div>
+{% if theme == 'halloween' %}
```
Comment (JennicaMistry) [inline views/components/footer.njk:1]: You have three footers, a halloween one, a christmas one and a normal one. You should split this out into multiple files and componentize it.

File: views/css/style.css [hunk 2]
```diff
@@ -1,9 +1,27 @@
 body {
     color: #252D4F;
 }
+
+/* ---- Halloween Scrollbar Styling */
+
+.halloween::-webkit-scrollbar
+{
```
Comment (JennicaMistry) [inline views/css/style.css:8]: Split out halloween and christmas styles into their own css files and so only load them when needed.

File: views/index.njk [hunk 3]
```diff
@@ -5,36 +5,78 @@
 {% endblock %}
 
 {% block content %}
-      <section class="hero d-flex justify-content-center mt-5">
+{% if theme == 'christmas' %}
```
Comment (JennicaMistry) [inline views/index.njk:8]: Componentize this into different views, one for christmas, halloween and default

File: src/app.ts [hunk 4]
```diff
@@ -6,6 +6,8 @@ import { dateFilter } from "./filters/DateFilter";
 import { unauthenticatedRouter } from "./routes/UnauthenticatedRouter";
 import { userRouter } from "./routes/UserRouter";
 import { setRoleInLocals } from "./middleware/SetLocalRoleMiddleware";
+import dotenv from "dotenv";
+dotenv.config();
```
[the remark this answers is not in the document]
Comment (AZZohaib) [reply src/app.ts:10]: Done

File: views/css/style.css (hunk 2 above)
Comment (AZZohaib) [reply views/css/style.css:8]: Done

File: views/components/navbar.njk [hunk 5]
```diff
@@ -1,20 +1,33 @@
-<nav class="navbar navbar-expand-lg navbar-light px-4 pt-4 pb-3">
-    <div class="nav-logo-container">
-        <a class="nav-item nav-link fw-medium" href="./"><img src="../assets/index/kainos-logo.png" alt="transparent kainos logo for the navbar" class="img-fluid"></a>
-    </div>
+<nav class="halloween-navbar navbar navbar-expand-lg navbar-light px-4 pt-4 pb-3">
```
[the remark this answers is not in the document]
Comment (AZZohaib) [reply views/components/navbar.njk:1]: Done

File: views/components/footer.njk (hunk 1 above)
Comment (AZZohaib) [reply views/components/footer.njk:1]: Done

File: src/app.ts [hunk 6]
```diff
@@ -30,10 +32,20 @@ app.use(session({ name:'kainos-job-roles', secret: 'SUPER_SECRET', cookie: { max
 
 declare module "express-session" {
   interface SessionData {
-    token: string;
+    token: string,
+    theme: string;
```
[the remark this answers is not in the document]
Comment (AZZohaib) [reply src/app.ts:36]: Done

File: views/index.njk (hunk 3 above)
Comment (AZZohaib) [reply views/index.njk:8]: done

File: views/jobRoles.njk [hunk 7]
```diff
@@ -13,11 +13,26 @@
         <div class="container">
 
             <div class="mainBody">
-                <h1 class = "m-0 p-0 d-flex flex-row justify-content-center mb-3">Job Roles List</h1>
+            {% if theme == 'halloween' %}
+            <div class="d-flex justify-content-center w-25 mx-auto gap-2 align-items-center">
```
[the remark this answers is not in the document]
Comment (AZZohaib) [reply views/jobRoles.njk:17]: Done

File: views/layouts/main.njk [hunk 8]
```diff
@@ -10,15 +10,17 @@
     </title>
     <link rel="shortcut icon" href="../assets/index/favicon.webp" type="image/x-icon">
     <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH" crossorigin="anonymous">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
```
[the remark this answers is not in the document]
Comment (AZZohaib) [reply views/layouts/main.njk:13]: Done

File: src/app.ts [hunk 9]
```diff
@@ -42,3 +54,10 @@ app.use(setRoleInLocals);
 app.use('/', userRouter);
 
 app.use('/', unauthenticatedRouter);
+
+// Changing theme
+app.use("/theme/:theme", (req, res) => {
```
[the remark this answers is not in the document]
Comment (AZZohaib) [reply src/app.ts:59]: Done

File: src/app.ts [hunk 10]
```diff
@@ -30,10 +32,20 @@ app.use(session({ name:'kainos-job-roles', secret: 'SUPER_SECRET', cookie: { max
...
 declare module "express-session" {
   interface SessionData {
-    token: string;
+    token: string,
+    theme: string;
   }
 }
 
+// Setting the theme in the session
+app.use((req, res, next) => {
```
[the remark this answers is not in the document]
Comment (AZZohaib) [reply src/app.ts:41]: Done

**Review by JennicaMistry — DISMISSED** [review]

**Review by Adnanhoq — APPROVED** [review]

**Review by JennicaMistry — APPROVED** [review]

**Review by Adnanhoq — APPROVED** [review]

**Pull request merged**